Modules/Filtering/ImageGrid/test/itkSliceBySliceImageFilterTest.cxx

```diff
@@ -165,13 +165,11 @@ itkSliceBySliceImageFilterTest(int argc, char * argv[])
   // spacing. We are setting the input image to have a non-zero
   // starting index.
   //
-  auto image = ImageType::New();
-  {
+  auto image = ImageType::CreateInitialized([&reader] {
     ImageType::RegionType region = reader->GetOutput()->GetLargestPossibleRegion();
     region.SetIndex(0, 10);
-    image->SetRegions(region);
-    image->AllocateInitialized();
-  }
+    return region;
+  }());
```

Comment:
FYI, this was probably the most interesting use case that I found. In the original code, the variable `region` was already kept inside a very small scope, between `{` and `}`. The proposed pull request places the `region` variable inside a lambda, in order to keep its scope small.